scripts/randdm

```diff
@@ -0,0 +1,156 @@
+    """
+    Generate random distance matrix. By default, generates a random distance
+    matrix drawing from a gaussian probability distribution. Since realistic
+    microbiome OTU tables and distance matrices typically do not follow such
+    a probability distribution, as they often contain "structure", i.e.
+    discrete block-like patterns or patterns along a gradient, another
+    option is provided to generate more realistic microbiome data that mimics
+    block or band-like structure, e.g. block-like could represent one grou of
+    microbes associated with a diseased state and another set associated with a
+    healthy state.
+
+    Parameters
+    ----------
+    dimensions : number
+        Dimension of the input matrix/matrices to generate.
+    output_dir : str
+        Path to the directory to output results to.
+    seed : number
+        Random seed for generating distance matrices. Optional.
+    subsample_dims : list of str
+        Generate subsampled distance matrix with given dimension
+        (must be smaller than original dimension) for each randomly
+        generated distance matrix. Can specify as int denoting specific
+        dimension or as percentage of, for example '25%', denoting percentage
+        of each main dimension specified in the 'dimensions' parameter.
+    structure : bool
+        Optional. Generate structure in the random data that
+        models distributions of microbial communities.
+        Options: "band" or "block". If not specified, then
+        draws from gaussian distribution to generate distance
+        matrices.
+    overwrite : bool
+        False by default. If True, overwrites output_dir if it already exists.
+
+    """
+
+    if seed is not None:
+        np.random.seed(seed)
+
+    if (not overwrite and os.path.exists(output_dir) and not click.confirm(
+            'The output directory %s exists. '
+            'Do you want to overwrite?' % output_dir)):
+        click.echo('Aborting.')
+        return
+
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    for dim in dimensions:
+        outpath = os.path.join(output_dir, 'randdm-{}.txt'.format(dim))
+
+        click.echo('Generating random distance matrix: %s' % outpath)
+
+        # Generate random distance matrix
+        if structure is not None:
+            click.echo('Generating synthetic OTU table with "{}"-like '
+                       'microbial distribution...'.format(structure))
+            if structure == 'band':
+                biom_table = band_table(dim, dim, seed=seed)[0]
+            elif structure == 'block':
+                biom_table = block_table(dim, dim, seed=seed)[0]
+            else:
+                raise ValueError('Invalid value for --structure parameter.')
+
+            otu_table = biom_table.matrix_data.todense()
+
+            click.echo('Generating distance matrix from OTU table...')
+            distance_matrix = beta_diversity('braycurtis', otu_table,
```

Comment:
Are we ok with hard-coding the distance measure?

Reply:
I changed it to be a CLI param with braycurtis as default value.